go-controller/pkg/config/config.go

```diff
@@ -909,6 +913,13 @@ var OVNGatewayFlags = []cli.Flag{
 		Destination: &cliConfig.Gateway.V6JoinSubnet,
 		Value:       Gateway.V6JoinSubnet,
 	},
+	&cli.StringFlag{
+		Name: "gateway-router-subnet",
+		Usage: "The Subnet to be used for the gateway router external port (shared mode only). " +
+			"auto-detected if not given",
+		Destination: &cliConfig.Gateway.RouterSubnet,
+		Value:       Gateway.RouterSubnet,
```

Comment:
What is the use case to override SmartNIC router subnet?

Reply:
@adrianchiris We need this RouterSubnet feature since we put BF2 x86 Host and BF2 Arm on separate subnets and VLANs.

Reply:
I see, so best to keep this then.